runtime/io.c

```diff
@@ -650,6 +650,16 @@ CAMLprim value caml_ml_set_binary_mode(value vchannel, value mode)
   return Val_unit;
 }
 
+CAMLprim value caml_ml_set_channel_unbuffered(value vchannel, value mode)
+{
+  struct channel * channel = Channel(vchannel);
+  if (Bool_val(mode))
+    channel->flags |= CHANNEL_FLAG_UNBUFFERED;
```

Comment:
You should also flush the channel at this point. Otherwise, if the channel contains buffered data, it will not be visible until the next output operation on the channel, which can be confusing to users.

Reply:
Good catch, fixed.